var/spack/repos/builtin/packages/hypre/package.py

```diff
@@ -46,7 +46,7 @@ class Hypre(Package):
     version('2.10.0b', '768be38793a35bb5d055905b271f5b8e')
     version('xsdk-0.2.0', tag='xsdk-0.2.0')
 
-    # hypre does not know how to build shared libraries on Darwin
+    # Older versions of hypre cannot build shared libraries on Darwin
```

Comment:
can you be more precise? When hypre learned this?

Reply:
@healther I wrote the patch with this functionality and submitted it to upstream, so hypre can build shared libraries on macOS with this patch as of September 11, 2018, 17:46, GMT-7.

The patch applies cleanly to versions 2.13.0 and later, which is why I chose to restrict the patch to that version range.

Reply:
But it is only needed for an even more restricted set of installations, i.e. only under darwin and only if you want shared libraries. I _think_ the policy is to be as restrictive as possible with patches, but at the same time I don't really care

Do you have a version for which the patch should be included upstream?

Reply:
Sorry I only saw the 
```
# Older versions of hypre cannot build shared libraries on Darwin
```
I'd prefer something like
```
# Versions prior to x.x.x cannot build shared libraries on Darwin, since 2.13.0 this patch applies, it is becoming part of hypre at y.y.y
```
If you know the values for `x.x.x` and `y.y.y`